pulsar-broker/src/main/java/org/apache/pulsar/EmbeddedPulsar.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar;
+
+import org.apache.pulsar.broker.ServiceConfiguration;
+import org.apache.pulsar.broker.ServiceConfigurationUtils;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class EmbeddedPulsar extends StandalonePulsarBase {
```

Comment:
why not just promote `EmbeddedPulsarBuilder` as `StandalonePulsarBuilder`? so make it a programming entrypoint for applications to use it to build a standalone pulsar cluster and use it in their applications.

Reply:
You mean change the name ?

Reply:
I mean 

- `PulsarStandaloneBase` doesn't have to be abstract. Let's rename it to `PulsarStandalone`.
- `PulsarStandaloneStarter` is a main program that accepts commandline args and starts `PulsarStandalone`
- Rename this class to `PulsarStandaloneBuilder`, which provides a builder function that build a `PulsarStandalone`.

`PulsarStandalone` itself is already an `embedded` cluster, there is no need to have a new name for it.